package-tests/test_content_type.py

```diff
@@ -46,7 +49,79 @@ def test_prepare_renderer():
     content_type.prepare_renderer(configurator)
 
 
+def test_check_global_allow_true():
+    """Test global_allow set to True."""
+    configurator = Configurator(
+        template='bobtemplates.plone:content_type',
+        target_directory='collective.foo.bar',
+        bobconfig={
+            'non_interactive': True,
+        },
+        variables={
+            'dexterity_type_global_allow': 'y',
+        },
+    )
+    with pytest.raises(SkipQuestion):
+        content_type.check_global_allow(configurator, None)
+
+
+def test_check_global_allow_false():
+    """Test global_allow set to False."""
+    configurator = Configurator(
+        template='bobtemplates.plone:content_type',
+        target_directory='collective.foo.bar',
+        bobconfig={
+            'non_interactive': True,
+        },
+        variables={
+            'dexterity_type_global_allow': 'f',
+            'dexterity_parent_container_type_name': 'Folder',
+        },
+    )
+    with pytest.raises(SkipQuestion):
+        content_type.check_global_allow(configurator, None)
+
+
+def test_update_ct_types_fti_xml(tmpdir):
+    """Test xml changes when changes are already in place."""
+    target_path = tmpdir.strpath + '/collective.sample'
+    package_path = target_path + '/src/collective/sample'
+    profiles_path = package_path + '/profiles/default/types'
+    os.makedirs(target_path)
+    os.makedirs(package_path)
+    os.makedirs(profiles_path)
+    template = """<?xml version="1.0"?>
+<object xmlns:i18n="http://xml.zope.org/namespaces/i18n"
+    name="parent"
+    meta_type="Dexterity FTI"
+    i18n:domain="collective.sample">
+  <property name="global_allow">True</property>
+  <property name="filter_content_types">True</property>
+  <property name="allowed_content_types">
+    <element value="child" />
+  </property>
+</object>
+"""
+    with open(os.path.join(profiles_path + '/parent.xml'), 'w') as f:
+        f.write(template)
+    configurator = Configurator(
+        template='bobtemplates.plone:content_type',
+        target_directory='collective.sample',
+        bobconfig={
+            'non_interactive': True,
+        },
+        variables={
+            'dexterity_type_name': 'child',
+            'dexterity_type_global_allow': 'f',
+            'dexterity_parent_container_type_name': 'parent',
+        },
+    )
+    configurator.variables['package_folder'] = package_path
+    content_type._update_ct_types_fti_xml(configurator)
```

Comment:
let's check the generated xml here, so that we are sure that the changes are made correctly.
This will also improve the test coverage.